topi/include/topi/transform.h

```diff
@@ -365,6 +365,84 @@ inline Array<Tensor> split(const Tensor& x,
   return result;
 }
 
+/*!
+* \brief strided_slice of a tensor
+*
+* \param x The input tensor
+* \param begin The indices to begin with in the slicing
+* \param end Indicies indicating end of the slice
+* \param strides Specifies the stride values, it can be negative
+* in that case, the input tensor will be reversed in that particular axis
+* \param name The name of the operation
+* \param tag The tag to mark the operation
+*
+* \return A Tensor whose op member is the split operation
+*/
+inline Tensor strided_slice(const Tensor& x,
+                            const Array<Expr>& begin,
+                            const Array<Expr>& end,
+                            const Array<Expr>& strides,
+                            std::string name = "tensor",
+                            std::string tag = kInjective) {
+  size_t src_tensor_dim = static_cast<size_t>(x->shape.size());
+  std::vector<int> begin_vec = GetConstIntValues(begin, "begin");
+  std::vector<int> end_vec = GetConstIntValues(end, "end");
+  std::vector<int> stride_vec = GetConstIntValues(strides, "strides");
+  // in case user has not provided begin indices for all the axes,
+  // then inflate it with default value = 0
+  for (size_t i = begin_vec.size(); i < src_tensor_dim; ++i) {
+    begin_vec.push_back(0);
+  }
+  // in case user has not provided end indices for all the axes,
+  // then inflate it with default value = input_tensor.shape[axis]
+  for (size_t i = end_vec.size(); i < src_tensor_dim; ++i) {
+    end_vec.push_back(GetConstInt(x->shape[i]));
+  }
+  // in case user has not provided stride values,
+  // then inflate it with default value = 1
+  for (size_t i = stride_vec.size(); i < src_tensor_dim; ++i) {
+    stride_vec.push_back(1);
+  }
+
+  Array<Expr> out_shape;
+
+  for (size_t i = 0; i < src_tensor_dim; ++i) {
+    int begin_range = stride_vec[i] < 0 ? -1 : 0;
+    int end_range = stride_vec[i] < 0 ? GetConstInt(x->shape[i]) - 1 : GetConstInt(x->shape[i]);
+    // transform negative indices to positive value, clips on the correct range
+    auto index_canonicalization = [x, i, begin_range, end_range](int index) {
+      if (index < 0) {
+        index += GetConstInt(x->shape[i]);
+      }
+      return std::min(std::max(index, begin_range), end_range);
+    };
+
+    int begin_i = begin_vec[i];
+    int end_i = end_vec[i];
+    begin_i = index_canonicalization(begin_i);
+    end_i = index_canonicalization(end_i);
+
```

Comment:
How about removing index_canonicalization()?  It looks simpler to me.
```
    int shape = GetConstInt(x->shape[i]);
    // transform negative indices to positive value                                                                                                                                  
    if (begin_i < 0) {
      begin_i += shape;
    }
    if (end_i < 0) {
      end_i += shape;
    }

    // clips on the correct range                                                                                                                                                    
    if (stride_vec[i] < 0) {
      begin_i = std::min(std::max(begin_i, -1), shape - 1);
      end_i = std::min(std::max(end_i, -1), shape - 1);
    } else {
      begin_i = std::min(std::max(begin_i, 0), shape);
      end_i = std::min(std::max(end_i, 0), shape);
    }
```

Reply:
@kazum, I prefer the function as it is more compact, also it's clear I think.

Please let me know your opinion.

Reply:
Okay, let's keep them.  I think it's a matter of taste.